routes/courses.js

```diff
@@ -1,13 +1,59 @@
 const express = require('express');
+const log = require('loglevel');
 const HttpError = require('http-errors');
-const { isEmpty } = require('./../services/utils');
+const { authorizeSession } = require('./../services/auth');
 const Course = require('./../models/Course');
 const User = require('./../models/User');
-const { authorizeSession } = require('./../services/auth');
+const { isEmpty } = require('./../services/utils');
 
 module.exports = () => {
   const router = express.Router();
 
+  // not currently used, but searches for a course based on id
+  router.get('/:id', authorizeSession, async (req, res, next) => {
+    try {
+      const id = req.params.id;
+      const course = await Course.findOneCourse({
+        id: id,
+      });
+      if (isEmpty(course)) {
+        throw new HttpError.NotFound();
+      }
+      log.info(`${req.method} ${req.originalUrl} success: returning course ${id}`);
+      return res.send(course);
+    } catch (error) {
+      next(error);
+    }
+  });
+
+  // needs the id of the user making the request for authorization
+  router.post('/', authorizeSession, async (req, res, next) => {
+    try {
+      const userId = res.locals.userId;
```

Comment:
@MattFriemoth the authentication is right here. Appears correct to me?

Reply:
Oh! You are correct. My mistake. Should have looked closer.